sonar-duplications/src/main/java/org/sonar/duplications/detector/original/OriginalCloneDetectionAlgorithm.java

```diff
@@ -213,9 +198,7 @@ private void reportClones(BlocksGroup beginGroup, BlocksGroup endGroup, int clon
         lastBlock.getEndLine());
 
       if (originResourceId.equals(part.getResourceId())) {
-        if (origin == null) {
-          origin = part;
-        } else if (part.getUnitStart() < origin.getUnitStart()) {
+        if (origin == null || part.getUnitStart() < origin.getUnitStart()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1066)